releasing.md

```diff
@@ -155,14 +155,46 @@ ks apply ${ENV} -c workflows
 Create a PR to update [jupyterhub_spawner.py](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/jupyterhub_spawner.py#L15) 
 to point to the newly built Jupyter notebook images.
 
-## Update the ksonnet configs
+## Create a release branch (if necessary)
 
-Create a PR to update the ksonnet configs to use the latest images built in the previous steps.
+If you aren't already working on a release branch (of the form `v${VER}-branch`, where `${VER}` is a major-minor version number), then create one.  Release branches serve several purposes:
 
-Presubmit tests should verify that the ksonnet configs work.
+1.  They allow a release wrangler or other developers to focus on a release without interrupting development on `master`,
+2.  they allow developers and sophisticated users to track the development of a release before a release candidate is declared, and
+3.  they simplify backporting critical bugfixes to a patchlevel release particular release stream (e.g., producing a `v0.1.1` from `v0.1-branch`), when appropriate.
 
-Submit the PR.
+### Release branching policy
 
-Wait for a passing postsubmit of the submitted PR.
+A release branch should be substantially _feature complete_ with respect to the intended release.  Code that is committed to `master` may be merged or cherry-picked on to a release branch, but code that is directly committed to the release branch should be solely applicable to that release (and should not be committed back to master).
 
-Tag the commit corresponding to the passing postsubmit with the appropriate tags for the release.
+## Updating the release branch and tagging a release
+
+0. Ensure that the release notes are up to date.
+1. Update the ksonnet configs on the release branch to use the latest images built in the previous steps.
+2. Presubmit tests should verify that the ksonnet configs work.
+3. Submit a PR to the release branch.
+4. Wait for a passing postsubmit of the submitted PR.  
+5. Once it passes, update this document on the release branch with specific image hashes.
```

Comment:
Why would we update with the image hashes? Shouldn't their be a single commit and that commit should pin all the images to specific images?

Reply:
Yes, let me explain:  when you put an explicit hash in `releasing.md`, I started to think it might make sense to have `releasing.md` on a release branch _actually contain_ the hashes used in the release, instead of placeholders.  This is of somewhat limited utility, but it does serve as accessible long-term documentation of what images were used in a particular release.